src/DefaultBuilder/src/StaticWebAssetsLoader.cs

```diff
@@ -0,0 +1,96 @@
+    {
+        internal const string StaticWebAssetsManifestName = "Microsoft.AspNetCore.StaticWebAssets.xml";
+
+        internal static void UseStaticWebAssets(IWebHostEnvironment environment)
+        {
+            using (var manifest = ResolveManifest(environment))
+            {
+                if (manifest != null)
+                {
+                    UseStaticWebAssetsCore(environment, manifest);
+                }
+            }
+        }
+
+        internal static void UseStaticWebAssetsCore(IWebHostEnvironment environment, Stream manifest)
+        {
+            var staticWebAssetsFileProvider = new List<IFileProvider>();
+            var webRootFileProvider = environment.WebRootFileProvider;
+
+            var additionalFiles = StaticWebAssetsReader.Parse(manifest)
+                .Select(cr => new StaticWebAssetsFileProvider(cr.BasePath, cr.Path))
+                .OfType<IFileProvider>() // Upcast so we can insert on the resulting list.
+                .ToList();
+
+            if (additionalFiles.Count == 0)
+            {
+                return;
+            }
+            else
+            {
+                additionalFiles.Insert(0, webRootFileProvider);
+                environment.WebRootFileProvider = new CompositeFileProvider(additionalFiles);
+            }
+        }
+
+        internal static Stream ResolveManifest(IWebHostEnvironment environment)
+        {
+            // We plan to remove the embedded file resolution code path in
+            // a future preview.
+            Assembly assembly = null;
+            try
+            {
+                assembly = Assembly.Load(environment.ApplicationName);
+            }
+            catch (Exception)
+            {
+            }
+
+            if (assembly != null && assembly.GetManifestResourceNames().Any(a => a == StaticWebAssetsManifestName))
+            {
+                return assembly.GetManifestResourceStream(StaticWebAssetsManifestName);
+            }
+            else
+            {
+                // Fallback to physical file as we plan to use a file on disk instead of the embedded resource.
+                var filePath = Path.Combine(Path.GetDirectoryName(GetAssemblyLocation(assembly)), $"{environment.ApplicationName}.StaticWebAssets.xml");
+                if (File.Exists(filePath))
+                {
+                    return File.OpenRead(filePath);
+                }
+                else
+                {
+                    // A missing manifest might simply mean that the feature is not enabled, so we simply
+                    // return early. Misconfigurations will be uncommon given that the entire process is automated
+                    // at build time.
+                    return null;
+                }
+            }
+        }
+
+        internal static string GetAssemblyLocation(Assembly assembly)
+        {
+            if (Uri.TryCreate(assembly.CodeBase, UriKind.Absolute, out var result) &&
```

Comment:
Why? Just use the location?

Reply:
Because when you run tests and do things like shadow-copying (like in tests), the location points to the wrong place. This is what MVC does to find related assemblies as part of application parts, and it has been improved a couple of times based on user feedback.

Reply:
So you’re telling me everywhere we use location is incorrect?

Reply:
@davidfowl It is correct in most cases, but it doesn't work with things like shadow-copying. I posted a link above to an MSDN blog explaining it, and as I said, this is the result of our own experience.